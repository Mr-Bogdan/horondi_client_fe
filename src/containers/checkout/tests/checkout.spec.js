import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Checkout from '../checkout';

jest.mock('react-redux');
jest.mock('../checkout.styles.js', () => ({ useStyles: () => ({}) }));
const mockUseContext = jest.fn().mockImplementation(() => ({
  isLight: true
}));
React.useContext = mockUseContext;

const dispatch = jest.fn();
const state = {
  language: 1,
  currency: 0,
  cartItems: [],
  deliveryType: '',
  loading: false,
  isOrderCreated: false,
  order: null
};

useDispatch.mockImplementation(() => dispatch);
useSelector.mockImplementation(() => state);

let wrapper;

describe('Checkout component tests', () => {
  beforeEach(() => {
    wrapper = shallow(<Checkout />);
  });

  afterEach(() => {
    wrapper = null;
  });

  it('Should render Checkout', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
