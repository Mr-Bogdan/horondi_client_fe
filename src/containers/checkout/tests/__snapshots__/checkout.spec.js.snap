// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThanksPage component tests should match snapshot 1`] = `
<div>
  <Redirect
    to="/"
  />
  <div>
    <CheckoutForm
      cartItems={Array []}
      currency={0}
      deliveryType=""
      errors={Object {}}
      isLightTheme={
        Object {
          "isLight": true,
        }
      }
      language={1}
      touched={Object {}}
      values={Object {}}
    />
  </div>
</div>
`;
